src/main/java/core/basesyntax/strategy/TransactionActivity.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.StorageTransaction;
+import core.basesyntax.strategy.impl.BalanceActivity;
+import core.basesyntax.strategy.impl.PurchaseActivity;
+import core.basesyntax.strategy.impl.ReturnActivity;
+import core.basesyntax.strategy.impl.SupplyActivity;
+
+public class TransactionActivity {
+    public StoreActivities getActivity(StorageTransaction transaction) {
+        switch (transaction.getOperation()) {
+            case BALANCE:
+                return new BalanceActivity();
+            case RETURN:
+                return new ReturnActivity();
+            case PURCHASE:
+                return new PurchaseActivity();
+            default:
+                return new SupplyActivity();
+        }
+    }
+}
```

Comment:
It is better to rewrite it like this
```suggestion
public class OperationStrategyImpl implements OperationStrategy {
    private Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;

    public OperationStrategyImpl(Map<FruitTransaction.Operation,
            OperationHandler> operationHandlerMap) {
        this.operationHandlerMap = operationHandlerMap;
    }

    @Override
    public OperationHandler get(FruitTransaction.Operation operation) {
        return operationHandlerMap.get(operation);
    }
}
```
so as not to violate solid principles